Code2Gether-Discord-Bot.Library/BusinessLogic/LeetLogic.cs

```diff
@@ -0,0 +1,100 @@
+            public JSONRequest(string qName)
+            {
+                variables.titleSlug = qName;
+            }
+        }
+        public class Question
+        {
+            public string questionId;
+            public string questionFrontendId;
+            public object boundTopicId;
+            public string title;
+            public string titleSlug;
+            public string content;
+        }
+        public class Variables
+        {
+           public string titleSlug;
+          
+        }
+        public LeetLogic(ILogger logger, ICommandContext context, string args)
+        {
+            _logger = logger;
+            _context = context;
+            _args = args;
+        }
+
+        public async Task<Embed> ExecuteAsync()
+        {
+            _logger.Log(_context);
+
+          
+            var result =  await getQuestions(_args.Replace("-", "").ToLower());
+            // Set from a private method
+            string title = result.Item1;    // Title for leet question
+            string description = result.Item2;  // Prompt for leet question
+            var embed = new EmbedBuilder()
+                .WithColor(Color.Purple)
+                .WithTitle($"Leet Question: {title}")
+                .WithDescription(description)
+                .WithAuthor(_context.Message.Author)
+                .Build();
+
+            return await Task.FromResult(embed);
+        }
+       public async Task<(string, string)> getQuestions(string qName)
+        {
+                TextInfo textInfo = new CultureInfo("en-US", false).TextInfo;
+                var title = textInfo.ToTitleCase(qName.Replace("-", " "));
+                var formattedQues = qName.Replace(" ", "-").ToLower();
+                var values = new JSONRequest(formattedQues);
+                var json = JsonConvert.SerializeObject(values);
+
+                var data = new StringContent(json, Encoding.UTF8, "application/json");
+
+                var url = "https://leetcode.com/graphql";
+                using var client = new HttpClient();
+
+                var response = await client.PostAsync(url, data);
+
+                string result = response.Content.ReadAsStringAsync().Result;
+              dynamic  desJson = JsonConvert.DeserializeObject(result);
+                var re = new Regex(@"<[^>]*>");
+
+               string plainResponse = desJson.data.question.content;
```

Comment:
Since the referenced variable is `dynamic`, `.data` can be null or nonexistent and suddenly exit the thread (happened during testing a nonexistent question.

Reply:
![image](https://user-images.githubusercontent.com/58752614/102633069-ecf0e780-411d-11eb-9417-c566a14693ed.png)